Add register.sh script and call from install.py

danjoneslf opened the pull request:
+ update readme to explain registration actions.

danjoneslf committed:
Add register.sh script and call from install.py - update readme to
explain registration actions.

gds-ahine reviewed (approved):
👍

0atman commented:
```shell
./install.py 

✔️ Detect-secrets hook installed
Enter your github username:
0atman
Performing GitHub OAuth
This creates a personal access token with read:user and read:org scopes. 
Please enter your GitHub 2FA code:
338004
Requesting authorization from GitHub - You will be prompted for your GitHub password.
Enter host password for user '0atman':
Submit registration request.
You have been registered successfully.
✔️ User registered
```
:+1:

alice-carr commented on register.sh:
```diff
@@ -0,0 +1,76 @@
+if [[ $response_type == "object" ]]; then
+  secret=$(echo $response | jq -r '.user_secret')
+  username=$(echo $response | jq -r '.username')
+
```
Mine failed first time round as I don't have jq installed - and we can't guarantee that everyone will have jq installed / or want to install it

0atman replied:
+1 for this, it's a shame to use jq, as the rest of this repo only relies on `python` and `curl`, I think.

0atman replied:
I have great news, we can do without it! 

```
cat my.json | python -m json.tool
```
Python has a built-in tool for pretty-printing json, which will exit with a non-zero code if it's invaild! :ok_hand:

0atman replied:
oh, but you can't slice off the secret and username with that... python rewrite?

0atman commented on install.py:
```diff
@@ -34,3 +34,13 @@ def _subprocess_call(cmd):  # this is the python 2.7 implementation
+_subprocess_call(
+    [
+        "bash",
+        os.environ["HOME"] + "/.gds-pre-commit/register.sh",
```
this no longer is how this file is used, @akinnane and @OllieJC made the repo relocatable

danjoneslf committed:
Add a check to make sure the GitHub authorization has return a JSON
response

danjoneslf committed:
Refactor register shell script into python.

danjoneslf committed:
Switch back to working bash version and add a main method to the
register.py script to enable debugging.

danjoneslf committed:
Implement register in python.
Note: try catch for input fails flake8.
Simplified run() function for running shell commands.
Removed incorrect reference to pdb from Pipfile.

0atman committed:
fixed linting errors on raw_input, blackened code

0atman commented on register.py:
```diff
@@ -0,0 +1,101 @@
+try:
+    from past.builtins import raw_input  # type: ignore
+except ImportError:
+    pass
```
@danjoneslf I've fixed it in the way I think is best. This'll fail python 2 syntax check, but - critically - pass our dev flake8!
